### PR TITLE
Jupyterlite

### PR DESCRIPTION
**Pull request by bnmajor**
Fixes the code path for now to avoid background jobs/threading in JupyterLite.

### CODE REVIEW

File: itkwidgets/integrations/environment.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    try:
+        from IPython import get_ipython
+        parent_header = get_ipython().parent_header
+        username = parent_header['header']['username']
+        if username == '':
+            IN_JUPYTERLAB = True
+        elif username == 'username':
+            IN_JUPYTER_NOTEBOOK = True
+        else:
+            IN_AWS = True
```
Comment (bnmajor) [inline itkwidgets/integrations/environment.py:22]: This condition has not been tested as I am still unable to access Studio Lab, all other conditions have.

File: itkwidgets/integrations/environment.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+IN_JUPYTER_NOTEBOOK = False
+IN_JUPYTERLAB = False
+IN_JUPYTERLITE = False
+IN_AWS = False
```
Comment (PaulHax) [inline itkwidgets/integrations/environment.py:4]: If there can only be 1 as true, an Enum would enforce that.
```
def findEnv():
  if something:
    return Env.JUPYTERLAB
  if anotherThing:
   return Env.JUPYTERLITE

env = findEnv()
```

**Review by thewtex — CHANGES_REQUESTED** [review]
Awesome!!

File: itkwidgets/integrations/environment.py (hunk 2 above)
Comment (bnmajor) [reply itkwidgets/integrations/environment.py]: Updated, thanks!

File: itkwidgets/integrations/environment.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        parent_header = get_ipython().parent_header
+        username = parent_header['header']['username']
+        if username == '':
+            IN_JUPYTERLAB = True
+        elif username == 'username':
+            IN_JUPYTER_NOTEBOOK = True
+        else:
+            IN_AWS = True
+    except AttributeError:
+        IN_JUPYTERLITE = True
```
[the remark this answers is not in the document]
Comment (bnmajor) [reply itkwidgets/integrations/environment.py]: Fixed!

File: itkwidgets/viewer.py [hunk 4]
```diff
@@ -101,11 +103,13 @@ async def run(self, ctx):
...
-        itk_viewer.registerEventListener(
-            'renderedImageAssigned', self.set_event
-        )
-        # Once the viewer has been created any queued requests can be run
-        asyncio.get_running_loop().call_soon_threadsafe(self.viewer_event.set)
+        if not IN_JUPYTERLITE:
+            itk_viewer.registerEventListener(
+                'renderedImageAssigned', self.set_event
+            )
+        if not IN_JUPYTERLITE:
```
[the remark this answers is not in the document]
Comment (bnmajor) [reply itkwidgets/viewer.py]: Oops! Yes, thank you, fixed now...

**Review by thewtex — APPROVED** [review]
Awesome, awesome!

**Pull request merged**